graphs/util/gpml_util.py

```diff
@@ -0,0 +1,284 @@
+import json
+
+
+def parse_gpml(graph_json, title):
+    parse_json = {"graph": {"edges": [], "nodes": []}, "metadata": {}}
+    # Use xml parser to parse the GPML file
+    import xml.etree.ElementTree as ET
+    tree = ET.fromstring(graph_json)
+    # The current GPML format
+    GPML = '{http://pathvisio.org/GPML/2013a}'
+    # Keep the overall count of nodes and create unique ids for
+    # datanodes, shape, labels.
+    # IMPORTANT!!!
+    count = 0
+
+    # Parsing the GPML file for Shape elements
+    shapes = tree.findall(GPML+'Shape')
+    parse_json, count = parse_shapes(parse_json, count, shapes)
+    # Find all the DataNodes in the GPML file which is used to draw various
+    # the various nodes of the graph
+    datanodes = tree.findall(GPML+'DataNode')
+    parse_json, count = parse_datanodes(parse_json, count, datanodes)
+    # Parsing the GPML file for Shape elements
+    labels = tree.findall(GPML+'Label')
+    parse_json, count = parse_labels(parse_json, count, labels)
+    # This is a crude version of getting interactions (edges) working
+    # Look for all the interactions
+    interactions = tree.findall(GPML+'Interaction')
+    parse_json, count = parse_interactions(parse_json, count, interactions)
+
+    layout = set_gpml_layout(parse_json)
+
+    # To add nodes to the group we parse through the group
+    # elements of the GPML file.
+    groups = tree.findall(GPML+'Group')
+    parse_json = parse_groups(parse_json, groups)
+    parse_json['metadata']['name'] = tree.attrib['Name'] or "GPML"
+
+    # No tags or description since CYJS doesn't give me any
+    parse_json['metadata']['tags'] = []
+    parse_json['metadata']['description'] = ""
+    title = title or parse_json['metadata']['name']
+    return parse_json, layout, title
+
+
+def parse_shapes(parse_json, count, shapes):
+    # Create a list of details of all the shape
+    shape_details = []
```

Comment:
Is "Details" a term used in GPML? If not, is "attributes" the right term. Use the right term here so tha tyou do not invent new terminology.